### ISSUE TITLE
[Bug]: 409 error when syncing to seafile (should also be the case for other strict webdav servers)

### ISSUE THREAD

**Issue by tmerten** [remotely-save/remotely-save#344]
### What happened?

Hi there,

I think I discovered the following issue. 

**TL;DR version**: 

There seems to be a bug where remotely-save does not create all parent folders on subsequent syncs which results in `409` errors with webdav.

**Long version**:

When I use remotely-save for the first time everything syncs exceptionally well to my Seafile server. After creating more folders and notes and syncing again I receive `409` (conflict) errors.

After a bit of debugging I noticed that this is due to missing folders. Here is what happens:

- When syncing for the first time all folders are created, then all notes are created
- Then I create new folders with notes in them locally
- When syncing again, remotely-save tries to create the file
  - *I think* it does not (always) create the parent folder explicitly
  - The webdav server responds with a 409 ([as expected](https://datatracker.ietf.org/doc/html/rfc4918#section-9.7.1), via [this sabredav issue](https://github.com/sabre-io/dav/issues/1304)) since it is not allowed to create a file if the containing folder is missing
- When I create the missing folders manually and continue syncing, everything works as expected

### What OS are you using?

Linux

### What remote cloud services are you using?

webdav

### Version of the plugin

0.3.25

### Version of Obsidian

1.3.5

### Using password or not

- [ ] Yes.

### Ensure no sensitive information

- [X] I ensure that no sensitive information is submitted in the issue.

**Comment by ksdavidc** [comment]
I appear to be getting the same issue with Dropbox

**Comment by ksdavidc** [comment]
sorry, perhaps not. By removing some recent files from the vault I stopped getting the problem ( a white screen of death). .Putting them back returned the white screen, but without the 409 error.

**Comment by tmerten** [comment]
I was able to test this on a nextcloud server as well. No problems there, maybe Seafile is more picky.

**Comment by fyears** [comment]
Are there any public seafile server or service i could test on? I am not using seafile by myself so it is gard to debug…

**Comment by fyears** [comment]
maybe you are using concurrency = 1 and there are some bugs. fixed in 0.3.38
